ledger/src/blockstore_db.rs

```diff
@@ -305,8 +307,40 @@ impl Rocks {
             new_cf_descriptor::<BankHash>(&access_type, &oldest_slot),
             new_cf_descriptor::<Root>(&access_type, &oldest_slot),
             new_cf_descriptor::<Index>(&access_type, &oldest_slot),
-            new_cf_descriptor::<ShredData>(&access_type, &oldest_slot),
-            new_cf_descriptor::<ShredCode>(&access_type, &oldest_slot),
+            match options.shred_storage_type {
+                ShredStorageType::RocksLevel => {
```

Comment:
The option to switch the compaction type should be plumbed as an option to the `ValidatorConfig` https://github.com/solana-labs/solana/blob/c62f9839a2a3e27081edee46069b608542b9957e/core/src/validator.rs#L113

Reply:
Good to know!  Will follow up with a different PR for that part.

For this PR, let's first make the FIFO setting available in the code base as the first step.

Reply:
got it, that works